config/check_dependencies.py

```diff
@@ -0,0 +1,169 @@
+#  Copyright (c) 2015-2023 TraceTronic GmbH
+#
+#  SPDX-License-Identifier: BSD-3-Clause
+#
+#  SPDX-License-Identifier: BSD-3-Clause
+
+import getopt
+import json
+import jsonschema
+import sys
+import os
+
+from os.path import exists
+
+filename = "check_dependencies.py"
+
+COMPATIBLE_LICENSES = [
+    "MIT",
+    "GPLv2",
+    "The MIT License",
+    "Apache-2.0",
+    "Apache-1.1",
+    "PSF-2.0",
+    "BSD",
+    "BSD-2-Clause",
+    "BSD-3-Clause",
+    "GNU LGPL",
+    "GNU LGPL-2.1",
+    "GNU LGPL-3.0",
+    "CDDL-1.1",
+    "EDL-1.0",
+    "MPL-2.0-no-copyleft-exception",
+    "EPL-1.0",
+    "The JSON License"
+]
+
+
+class ComponentValidator:
+    def __init__(self, allow_component, sbom_component):
+        self.allow = allow_component
+        self.sbom = sbom_component
+
+        self.errors = []
+
+    def validate(self, product=False):
+        self.__name()
+        self.__licenses()
+        if not product:
+            self.__version()
+        return self.errors
+
+    def __name(self):
+        sbom_name = "{}:{}".format(self.sbom["group"], self.sbom["name"])
+        if self.allow["moduleName"] != sbom_name:
+            self.append_error("name", self.allow["moduleName"], sbom_name)
```

Comment:
ComponentValidator is only instanciated, when the names match, so this check will be fine all the time